src/templates/post.js

```diff
@@ -75,7 +75,7 @@ const PostPage = ({ data, pageContext }) => {
                 {(post.headerImage) && (
                     <meta
                         property="og:image"
-                        content={origin + post.headerImage[0].localFile.publicURL}
+                        content={(origin ?? data.site.siteMetadata.siteUrl) + post.headerImage[0].localFile.publicURL}
```

Comment:
The `origin` default value is `''` (default value in useState), so this condition is rendered as empty string (see:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator )

We should use `||` operator, or ignore `origin` value and use only `data.site.siteMetadata.siteUrl + post ...`